File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/main/java/simulacionBandaMusica/factories/GuitarristaFactory.java
package simulacionBandaMusica.factories;

import simulacionBandaMusica.enums.TipoGuitarra;
import simulacionBandaMusica.models.Guitarrista;

import static simulacionBandaMusica.utils.sorteoDatos.*;

public class GuitarristaFactory {

    public static Guitarrista create() {
        String name = randomName();
        int experienceYear = randomAnno();
        TipoGuitarra tipoGuitarra = randomTipoGuitarra();
        return new Guitarrista(name, "Guitarra", experienceYear, tipoGuitarra);
    }

}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/main/kotlin/simulacionBandaMusica/factories/GuitarristaFactory.kt
package simulacionBandaMusica.factories

import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.sorteoDatos

object GuitarristaFactory {
    fun create(): Musico.Guitarrista {
        val name = sorteoDatos.randomName()
        val experienceYear = sorteoDatos.randomAnno()
        val tipoGuitarra = sorteoDatos.randomTipoGuitarra()
        return Musico.Guitarrista(name, "Guitarra", experienceYear, tipoGuitarra)
    }
}